### PR TITLE
add cache response header feature

### PR DESCRIPTION
**Pull request by trx35479**
@anandkunal Please check my PR. I added cache response headers feature.

### CODE REVIEW

**Comment by anandkunal** [comment]
The code looks great. Can include updated documentation as well? Specifically looking for the menu addition at the top level `incapsula.erb` and the appropriate documentation in the `docs` directory: https://github.com/imperva/terraform-provider-incapsula/tree/master/website

**Comment by trx35479** [comment]
@anandkunal docs have been updated.

File: incapsula/client_cache_response_headers.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"io/ioutil"
+	"log"
+	"net/url"
+	"strconv"
+)
+
+// endpoints
+const advancedCacheEndpoint = "sites/performance/response-headers"
+
+// struc
```
Comment (anandkunal) [inline incapsula/client_cache_response_headers.go:15]: Spelling from struc to struct.

File: incapsula/client_cache_response_headers.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	Res        int    `json:"res"`
+	ResMessage string `json:"res_message"`
+	DebugInfo  struct {
+		IDInfo string `json:"id-info"`
+	} `json:"debug_info"`
+}
+
+// ConfigureAdvanceCache adds an WAF Sec rule
+func (c *Client) ConfigureAdvanceCache(siteID int, cacheHeaders, cacheAllHeaders string) (*AddCacheHeaderResponse, error) {
+	log.Printf("[INFO] Configuring Incapsula Advance Cacheing site id: %d\n", siteID)
```
Comment (anandkunal) [inline incapsula/client_cache_response_headers.go:26]: Spelling - should be Incapsula Advanced Caching

File: incapsula/provider.go [hunk 3]
```diff
@@ -70,7 +70,11 @@ func Provider() terraform.ResourceProvider {
 			"incapsula_data_center":             resourceDataCenter(),
 			"incapsula_data_center_server":      resourceDataCenterServer(),
 			"incapsula_custom_certificate":      resourceCertificate(),
+<<<<<<< HEAD
```
Comment (anandkunal) [inline incapsula/provider.go:73]: Merge issues in this file.

File: incapsula/resource_certificate.go [hunk 4]
```diff
@@ -130,3 +130,12 @@ func resourceCertificateDelete(d *schema.ResourceData, m interface{}) error {
 
 	return nil
 }
+<<<<<<< HEAD
```
Comment (anandkunal) [inline incapsula/resource_certificate.go:133]: Merge issues in this file.

**Comment by trx35479** [comment]
@anandkunal this needs more fixing since this PR was based on the old terraform library.

**Comment by anandkunal** [comment]
Hi @trx35479 - let me know when it's safe to review (looks like there are still conflicts). Let me know if you want me to help out with the merge - happy to jump in. I'll leave this PR open.

**Comment by trx35479** [comment]
@anandkunal I think I fixed all the merge issues. Could I have review on this?

**Comment by anandkunal** [comment]
Hi @trx35479 - I believe I landed what you're going for a few weeks ago directly on the `site` resource, in addition to other performance settings. Here are the keys from the docs:

* `perf_response_cache_response_header_mode` - (Optional) The working mode for caching response headers. Options are `all` and `custom`.
* `perf_response_cache_response_headers` - (Optional) An array of strings representing the response headers to be cached when working in `custom` mode. If empty, no response headers are cached.

Can you try this on the site resource and let me know what you think?

**Comment by anandkunal** [comment]
Hi @trx35479 - just a friendly ping to see if you managed to test the performance settings that landed in the site resource.

**Comment by trx35479** [comment]
@anandkunal I wasn't able to test it. If that's the case we can just close this PR not problem at all.

**Pull request closed without merging**